odoo/tests/common.py

```diff
@@ -348,6 +348,23 @@ def __init__(self, methodName='runTest'):
         self.addTypeEqualityFunc(etree._Element, self.assertTreesEqual)
         self.addTypeEqualityFunc(html.HtmlElement, self.assertTreesEqual)
 
+    def run(self, result):
+        ODOO_TEST_FAILURE_RETRIES = int(os.environ.get('ODOO_TEST_FAILURE_RETRIES', 1))
+        failure = False
+        ODOO_TEST_FAILURE_RETRIES = 2 # TODO remove me
+        for retry in range(ODOO_TEST_FAILURE_RETRIES):
+            if retry == 0 or failure:
+                if retry:
+                    _logger.runbot(f'Retrying a failed test: {self}')
+                if retry != ODOO_TEST_FAILURE_RETRIES-1:
+                    with result.soft_fail(), quiet_logger() as quiet_log:
+                        super().run(result)
+                        failure = result.had_failure or quiet_log.had_error_log
```

Comment:
> `or quiet_log.had_error_log`

I still don't think I agree with that: though it may (will?) fail the build on runbot, `_logger.error` does not fail the tests normally.